src/erlcloud_ddb2.erl

```diff
@@ -1574,8 +1584,10 @@ batch_get_item(RequestItems, Opts) ->
                             batch_get_item_return().
 batch_get_item(RequestItems, Opts, Config) ->
     {AwsOpts, DdbOpts} = opts(batch_get_item_opts(), Opts),
+    NoRequest = proplists:get_value(no_request, DdbOpts, false),
```

Comment:
This seems a bit repetitive, but I can't figure out how to have a better pattern, since `Opts` is never passed to `request` function. This is a general comment, not a request or suggestion for change.

Reply:
TBH I would suggest passing `DdbOpts` to `erlcloud_ddb_impl:request/4` as a 4th argument (and keep `erlcloud_ddb_impl:request/3` passing to `erlcloud_ddb_impl:request/4` with `DdbOpts = []`), then read the `no_request` option out there in exactly one place. That would make this substantially more concise and also flexible to do similar things in the future without changing the arity of every function call again.

Reply:
DdbOpts is now checked inside erlcloud_ddb_impl:request/4

Reply:
> TBH I would suggest passing

I was going to... but I didn't want to go that far 😄